src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1138,7 +1142,8 @@
                                     :environment_id         (ensure-environment environment)
                                     :status_id              (ensure-status status)}))]
                (->> resource_events
-                    (map (comp convert-containment-path #(assoc % :report_id id)))
+                    (map (comp convert-containment-path #(assoc % :report_id id
+                                                           :certname_id (certname-id certname))))
```

Comment:
Also I might just make two separate mappings one for `convert` and  one for `assoc` or just lose the `->>`